packages/piral-cli-webpack/src/configs/common.ts

```diff
@@ -87,28 +102,53 @@ export function getRules(baseDir: string): Array<RuleSetRule> {
       use: [styleLoader, 'css-loader'],
     },
     {
-      test: /\.m?jsx?$/i,
+      test: /\.m?[jt]sx?$/i,
       use: [
         {
           loader: 'babel-loader',
           options: {
-            presets: ['@babel/preset-env', '@babel/preset-react'],
+            presets: [
+              [
+                '@babel/preset-env',
+                {
+                  targets: {
+                    browsers: 'ie >= 9',
+                    node: '6.1.0',
+                    esmodules: false,
+                  },
+                  modules: false,
+                  debug: true,
+                  include: [],
+                  exclude: [],
+                  useBuiltIns: false,
+                  forceAllTransforms: false,
+                  shippedProposals: false,
+                },
+              ],
+              '@babel/preset-react',
+              // When ts-loader is used with `@babel/plugin-transform-runtime`, it fails to convert jsx well, for example: arrow function is not converted, so it is replaced with `@babel/preset-typescript`.
+              '@babel/preset-typescript',
+            ],
+            plugins: [
+              [
+                '@babel/plugin-transform-runtime',
+                {
+                  corejs: 3,
+                  helpers: true,
+                  regenerator: true,
+                  useESModules: false,
+                },
+              ],
+              '@babel/plugin-proposal-object-rest-spread',
+              '@babel/plugin-transform-modules-commonjs',
+              ['@babel/plugin-proposal-decorators', { legacy: true }],
+              ['@babel/plugin-proposal-class-properties', { loose: true }],
+            ],
           },
         },
       ],
       exclude: nodeModules,
     },
-    {
```

Comment:
Removing this I would then add the [fork-ts-checker-webpack-plugin](https://www.npmjs.com/package/fork-ts-checker-webpack-plugin).

Reply:
Do you mean to remove `@babel/preset-typescript`?
I haven't figured out why the ts-loader did not convert the arrow function of the component.

Reply:
I think its either ts-loader or babel with ts + the fork-ts-checker. The latter combo could be the best in terms of performance and flexibility.

Regarding the arrow function: Is it maybe due to the used tsconfig? Maybe the target there is ES6?

Reply:
Aha, you may be right, I mistaken the `tsconfig.json` in the `piral-cli-webpack` directory for the `tsconfig.json` in the shell application directory, and I will return after verifying that there is no problem submitted.

Reply:
The problem has been resolved and verified